stdlib/public/core/Sort.swift

```diff
@@ -516,35 +516,34 @@ internal func _siftDown<C: MutableCollection & RandomAccessCollection>(
   subRange range: Range<C.Index>,
   by areInIncreasingOrder: (C.Element, C.Element) throws -> Bool
 ) rethrows {
-
-  let countToIndex = elements.distance(from: range.lowerBound, to: index)
-  let countFromIndex = elements.distance(from: index, to: range.upperBound)
-  // Check if left child is within bounds. If not, return, because there are
+  var index = index
+  var countToIndex = elements.distance(from: range.lowerBound, to: index)
+  var countFromIndex = elements.distance(from: index, to: range.upperBound)
+  // Check if left child is within bounds. If not, stop iterating, because there are
   // no children of the given node in the heap.
-  if countToIndex + 1 >= countFromIndex {
-    return
-  }
-  let left = elements.index(index, offsetBy: countToIndex + 1)
-  var largest = index
-  if (try areInIncreasingOrder(elements[largest], elements[left])) {
-    largest = left
-  }
-  // Check if right child is also within bounds before trying to examine it.
-  if countToIndex + 2 < countFromIndex {
-    let right = elements.index(after: left)
-    if (try areInIncreasingOrder(elements[largest], elements[right])) {
-      largest = right
+  while countToIndex + 1 < countFromIndex {
+    let left = elements.index(index, offsetBy: countToIndex + 1)
+    var largest = index
+    if (try areInIncreasingOrder(elements[largest], elements[left])) {
+      largest = left
     }
-  }
-  // If a child is bigger than the current node, swap them and continue sifting
-  // down.
-  if largest != index {
-    elements.swapAt(index, largest)
-    try _siftDown(
-      &elements,
-      index: largest,
-      subRange: range
-      , by: areInIncreasingOrder)
+    // Check if right child is also within bounds before trying to examine it.
+    if countToIndex + 2 < countFromIndex {
+      let right = elements.index(after: left)
+      if (try areInIncreasingOrder(elements[largest], elements[right])) {
+        largest = right
+      }
+    }
+    // If a child is bigger than the current node, swap them and continue sifting
+    // down.
+    if largest != index {
+      elements.swapAt(index, largest)
+      index = largest
+    } else {
+      break
```

Comment:
it would be nice if the variable updates could move up into the `if` so the break is the last line. It'd be even nicer if this list could be recast as a `repeat/while` so the break isn't necessary, but maybe that's not easily done.